app/Http/Controllers/Plan/PlansController.php

```diff
@@ -566,10 +567,14 @@ public function storeDay(Request $request, $plan_id)
             ->where('plan_id', $plan->id)
             ->where('user_id', $user->id)
             ->get()->pluck('id');
-        $plan_days_data = $new_playlists->map(function ($item) use ($plan) {
+
+        $add_to_end = checkParam('add_to_end') ?? false;
+
+        $plan_days_data = $new_playlists->map(function ($item, $index) use ($plan, $add_to_end, $current_days_size) {
             return [
                 'plan_id'               => $plan->id,
                 'playlist_id'           => $item,
+                'order_column' => $add_to_end ? $current_days_size + ($index + 1) : 0,
```

Comment:
This is because I have recommended to keep the logic that we can see in the endpoint to update a plan (/app/Http/Controllers/Plan/PlansController.php line 375), where the index starts at 1.